kubectl-plugin/pkg/util/client/client_test.go

```diff
@@ -386,3 +390,83 @@ func TestGetRayHeadSvcNameByRayService(t *testing.T) {
 		})
 	}
 }
+
+type MockWatcher struct {
+	resultChan chan watch.Event
+	mock.Mock
+}
+
+func (m *MockWatcher) Stop() {
+	m.Called()
+	close(m.resultChan)
+}
+
+func (m *MockWatcher) ResultChan() <-chan watch.Event {
+	return m.resultChan
+}
```

Comment:
Why not to use [FakeWatcher](https://pkg.go.dev/k8s.io/apimachinery/pkg/watch#FakeWatcher) directly?

Reply:
I couldn't figure out how to use it. Do you have a code snippet?

Reply:
It implements `watch.Interface`, so use it as a regular watcher. Maybe let `WaitRayClusterProvisioned` take another argument, `watcher watch.Interface`?

https://github.com/kubernetes/apimachinery/blob/47e7fa9a40a229d501d130fe434ca63eadee94dc/pkg/watch/watch_test.go#L33-L78

Reply:
Thanks, I think I finally figured it out. Updated 😅